lib/content-negotiation.js

```diff
@@ -0,0 +1,80 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-error-handler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var accepts = require('accepts');
+var debug = require('debug')('strong-error-handler:http-response');
+var sendJson = require('./send-json');
+var sendHtml = require('./send-html');
+
+module.exports = negotiateContentProducer;
+
+/**
+ * Handles req.accepts and req.query._format and options.defaultTypes
+ * to resolve the correct operation
+ *
+ * @param req request object
+ * @param {Object} options options of strong-error-handler
+ * @returns {Function} Opeartion function with signature `fn(res, data)`
+ */
+function negotiateContentProducer(req, options) {
+  var SUPPORTED_TYPES = [
+    'application/json', 'json',
+    'text/html', 'html',
+  ];
+
+  options = options || {};
+  var defaultType = 'json';
+
+  // checking if user provided defaultType is supported
+  if (options.defaultType) {
+    if (SUPPORTED_TYPES.indexOf(options.defaultType) > -1) {
+      debug('Accepting options.defaultType `%s`', options.defaultType);
+      defaultType = options.defaultType;
+    } else {
+      console.warn('defaultType: `%s` is not supported,' +
```

Comment:
This will pollute console log a lot.

I think we should either use `debug` instead of `console.log`, or if you want to make this more user-friendly, then print the warning from `createStrongErrorHandler`.